gems/sorbet-runtime/test/types/props/_props.rb

```diff
@@ -408,7 +398,98 @@ class OverrideProps3 < OverrideProps
         end
       end
 
-      assert(error.message.include?("Attempted to override a prop :b on class Opus::Types::Test::Props::PropsTest::OverrideProps3 that doesn't already exist"))
+      assert(error.message.include?("You marked the getter for prop :b as `override`, but the method `b` doesn't exist to be overridden."))
+    end
+
+    module ManualGetter
+      extend T::Sig
+
+      sig { overridable.returns(Integer) }
+      def a; 0; end
+    end
+
+    module ManualSetter
+      extend T::Sig
+      extend T::Helpers
+
+      sig { overridable.params(a: Integer).returns(Integer) }
+      def a=(a); 0; end
+    end
+
+    it 'allows overriding normal methods' do
+      class OverrideProps4 < T::Struct
+        include ManualGetter
+        include ManualSetter
+
+        prop :a, Integer, override: true
+      end
+    end
+
+    it 'allows overriding only one half' do
+      class OverrideProps5 < T::Struct
+        include ManualGetter
+
+        prop :a, Integer, override: :reader
+      end
+    end
+
+    it 'permits {allow_incompatible: true}' do
+      class OverrideProps6 < T::Struct
+        include ManualSetter
+
+        prop :a, Integer, override: {writer: {allow_incompatible: true}}
+      end
+    end
+
+    it 'accepts override: false' do
+      class OverrideNothing < T::Struct
+        prop :a, Integer, override: false
+      end
+    end
+
+    # This came up with props defined by Chalk::ODM::AbstractModel
+    it 'handles indirect prop definitions' do
+      module A
+        extend T::Helpers
+        def self.included(base=nil, &block)
+          if base.nil?
+            raise "double-defined block" if instance_variable_defined?(:@_included_block)
+
+            @_included_block = block
+          else
+            super
+          end
+        end
+
+        def self.append_features(base)
+          return false if base < self
+          super
+          base.class_eval(&@_included_block) if instance_variable_defined?(:@_included_block)
+        end
+
+        included do
+          T.unsafe(self).prop :foo, T.nilable(String)
+        end
+      end
+
+      class B
+        include T::Props
+        include A
+        prop :foo, String, override: true
```

Comment:
```suggestion
      class B
        include T::Props
        prop :foo, T.nilable(String)
        prop :foo, String, override: true
```

Reply:
I believe this tests the same thing, but without any of the metaprogramming or misdirection that the `included do ... end` pattern brings.

Reply:
I think it ultimately tests the same functionality, but the point was partially to test the weird indirection with `prop` being called via `included`. Happy to simplify it if you don't think it's worth it, though.

Reply:
I very much want zero usages of `append_features` in sorbet-runtime, even in tests.